server/src/domain/plugin_backend_handler.rs

```diff
@@ -0,0 +1,510 @@
+    types::PluginContext,
+};
+use lldap_plugin_kv_store::store::PluginKeyValueStore;
+
+use crate::{
+    domain::{
+        ldap::utils::LdapInfo, opaque_handler::OpaqueHandler, plugin::backend::ServerBackendAPI,
+        sql_backend_handler::SqlBackendHandler,
+    },
+    infra::{ldap_handler::InternalSearchResults, tcp_backend_handler::TcpBackendHandler},
+};
+
+use tracing::instrument;
+
+#[derive(Clone, Debug)]
+pub struct PluginBackendHandler {
+    pub(crate) backend_handler: SqlBackendHandler,
+    pub(crate) backend_api: &'static ServerBackendAPI<SqlBackendHandler>,
+    pub(crate) plugin_handler: PluginHandler<PluginKeyValueStore>,
+    pub(crate) request_context: RequestContext,
+}
+
+impl PluginBackendHandler {
+    pub fn new(
+        backend_handler: &SqlBackendHandler,
+        plugin_handler: PluginHandler<PluginKeyValueStore>,
+    ) -> Self {
+        let api: &'static ServerBackendAPI<SqlBackendHandler> =
+            Box::leak(Box::new(ServerBackendAPI {
+                backend_handler: backend_handler.clone(),
+                ldap_info: LdapInfo::from(
+                    backend_handler.config.ldap_base_dn.clone(),
+                    backend_handler.config.ignored_user_attributes.clone(),
+                    backend_handler.config.ignored_group_attributes.clone(),
+                ),
+            }));
+        PluginBackendHandler {
+            backend_handler: backend_handler.clone(),
+            backend_api: api,
+            plugin_handler: plugin_handler,
+            request_context: RequestContext::empty(),
+        }
+    }
+    pub fn new_plugin_context(&self) -> PluginContext<ServerBackendAPI<SqlBackendHandler>> {
+        PluginContext::new(
+            self.backend_api,
+            self.request_context.validation_results.clone(),
+        )
+    }
+    pub async fn initialize_plugins(&self) -> std::result::Result<(), String> {
+        let ctx = self.new_plugin_context();
+        self.plugin_handler.initialize_plugins(ctx).await
+    }
+}
+
+#[async_trait]
+impl BackendHandler for PluginBackendHandler {}
+
+impl WithContextHandler for PluginBackendHandler {
+    fn with_context(&self, context: RequestContext) -> Self {
+        PluginBackendHandler {
+            backend_handler: self.backend_handler.clone(),
+            backend_api: self.backend_api,
+            plugin_handler: self.plugin_handler.clone(),
+            request_context: context,
+        }
+    }
+}
+
+#[async_trait]
+impl ReadSchemaBackendHandler for PluginBackendHandler {
+    #[instrument(skip(self), level = "debug", ret, err)]
+    async fn get_schema(&self) -> Result<Schema> {
+        self.backend_handler.get_schema().await
+    }
+}
+
+#[async_trait]
+impl SchemaBackendHandler for PluginBackendHandler {
+    #[instrument(skip(self), level = "debug", ret, err)]
+    async fn add_user_attribute(&self, request: CreateAttributeRequest) -> Result<()> {
+        self.backend_handler.add_user_attribute(request).await
```

Comment:
I suppose this will actually have plugin calls later on?

Reply:
Yes. I just wanted to get the structure ACK'ed before doing more. Seemed unnecessary, if they all needed to be changed, or something.